src/main/scala/uk/gov/hmrc/ct/ct600/v3/calculations/LoansToParticipatorsCalculator.scala

```diff
@@ -149,4 +168,47 @@ trait LoansToParticipatorsCalculator extends CtTypeConverters {
     case _ => B485(false)
   }
 
+  private def validRepaymentsWihin9Months(loans: List[Loan], date: LocalDate): List[Repayment] = for {
```

Comment:
As a standard we keep the calculations separate from the boxes when they are complicated...usually because you need to see and test the calculations across a section like L2P as a whole. You could argue moving them into the boxes individually if you wanted. They are sharing common code though. You can argue it many ways :)